src/main/java/org/c02e/jpgpj/JCAContextHelper.java

```diff
@@ -0,0 +1,111 @@
+package org.c02e.jpgpj;
+
+import org.bouncycastle.gpg.keybox.jcajce.JcaKeyBoxBuilder;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.operator.PGPDigestCalculatorProvider;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPContentVerifierBuilderProvider;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPDigestCalculatorProviderBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBEDataDecryptorFactoryBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePGPDataEncryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyDataDecryptorFactoryBuilder;
+
+import java.security.Provider;
+import java.security.Security;
+
+/**
+ * Helper class for Java Cryptography Architecture (JCA) context
+ * consisting of {@link java.security.Provider provider}
+ * Note: The following class is not thread safe, the security provider should not be changed during PGP operations
+ */
+public class JCAContextHelper {
+	private static Provider securityProvider = Security.getProvider("BC");
```

Comment:
For the convenience of users who don't otherwise need to fiddle with JCA, let's try to automatically set the `securityProvider` property to a `BouncyCastleProvider` instance by default -- and fail as gracefully as possible if it's not available. Something like this:
```
    private static Provider securityProvider;
    static {
        try {
           securityProvider = (Provider) Class.forName(
                "org.bouncycastle.jce.provider.BouncyCastleProvider"
            ).getDeclaredConstructor().newInstance();
        } catch (Exception e) {
            LoggerFactory.getLogger(JCAContextHelper.class.getName()).warn(
                "not using BC security provider");
        }
    }
```